ChatQuickstart/QuickstartChatManager.swift

```diff
@@ -0,0 +1,152 @@
+        messages.append(message)
+        
+        DispatchQueue.main.async() {
+            if let delegate = self.chatManagerDelegate {
+                delegate.reloadMessages()
+                if self.messages.count > 0 {
+                    delegate.receivedNewMessage()
+                }
+            }
+        }
+    }
+    
+    func sendMessage(_ messageText:String, completion: @escaping (TCHResult, TCHMessage?) -> Void) {
+        if let messages = self.channel?.messages {
+            let messageOptions = TCHMessageOptions().withBody(messageText)
+            messages.sendMessage(with: messageOptions, completion: { (result, message) in
+                completion(result, message)
+            })
+        }
+    }
+    
+    
+    func login(_ identity:String, completion:@escaping (Bool)->Void) {
+        // Fetch Access Token from the server and initialize Chat Client - this assumes you are
+        // calling a Twilio function, as described in the Quickstart docs
+        let urlString = "\(tokenURL)?identity=\(identity)"
+        
+        TokenUtils.retrieveToken(url: urlString) { (token, identity, error) in
+            guard let token = token else {
+                print("Error retrieving token: \(error.debugDescription)")
+                completion(false)
+                return
+            }
+            // Set up Twilio Chat client
+            TwilioChatClient.chatClient(withToken: token, properties: nil, delegate: self) {
+                (result, chatClient) in
+                self.client = chatClient
+                completion(result.isSuccessful())
+            }
+        }
+    }
+    
+    
+    func shutdown() {
+        if let client = client {
+            client.delegate = nil
+            client.shutdown()
+            self.client = nil
+        }
+    }
+    
+    private func createChannel(_ completion: @escaping () -> Void) {
+        guard let client = client, let channelsList = client.channelsList() else {
+            return
+        }
+        // Create the channel if it hasn't been created yet
+        let options = [
+            TCHChannelOptionUniqueName: uniqueChannelName,
+            TCHChannelOptionFriendlyName: friendlyChannelName,
+            TCHChannelOptionType: TCHChannelType.private.rawValue
+            ] as [String : Any]
+        channelsList.createChannel(options: options, completion: { channelResult, channel in
+            if (channelResult.isSuccessful()) {
+                print("Channel created.")
+            } else {
+                print("Channel NOT created.")
+            }
+            completion()
+        })
+        
+    }
+    
+    private func checkChannelCreation(_ completion: @escaping(Bool) -> Void) {
+        guard let client = client, let channelsList = client.channelsList() else {
+            return
+        }
+        channelsList.channel(withSidOrUniqueName: uniqueChannelName, completion: { (result, channel) in
```

Comment:
I would avoid calling channel() multiple times without a need. This causes a roundtrip to server.

Best would be to return Option<Channel> from checkChannelCreation (return None if no channel found, otherwise return the found Channel itself, which could then be passed to joinChannel).